packages/extension-starter/README.md

```diff
@@ -62,6 +62,39 @@ the extension as they like.
 > Congratulations! The Sample extension was successfully installed! Please visit https://www.npmjs.com/package/@salesforce/extension-starter for more information on how to use this extension.
 ```
 
+# State Management
+
+By default all extensions are enhanced with state management using the `withApplicationExtensionStore` higher-order component. Under the hood
+the state is provided using [Zustand](https://www.npmjs.com/package/zustand) as a global store for the entire PWA-Kit application. 
+Each Application Extension inserts a "slice" into this global store following the 
+[slicing pattern](https://github.com/pmndrs/zustand/blob/37e1e3f193a5e5dec6fbd0f07514aec59a187e01/docs/guides/slices-pattern.md). 
+This allows you to have data separation from one extension to the other when it's important, but also allows you to access state its 
+associated actions of other extensions when needed. 
```

Comment:
I think that we need to highlight that all slices in the global store are publicly accessible across extensions, any extension can read or modify the state of another extension, which helps extensions to interact but also can expose sensitive or private details.

We can say something like: The global store follows an open design, meaning all slices are accessible to every extension. If an extension requires private state or restricted access to sensitive data, consider implementing an access control mechanism using middleware or utility functions.

Reply:
Or maybe: Consider using Zustand Public State for a global store that is shared across extensions. Continue to utilize a private state for internal state management with local React useState or context within the extension.

Reply:
Good call out. I think I'll beef up that section a little. What I was trying to highlight here is that there is data isolation when you want it, e.g. using state within your extension, but there is also the ability to view and modify any extensions state via its actions from the global store.

I'll ping you when I have something a little better.

Reply:
updated